### ISSUE TITLE
whisper.InvalidTimeInterval from carbon-sync

### ISSUE THREAD

**Issue by garrettux** [graphite-project/carbonate#41]
Hi, we're trying to use carbon-sync, and are running into `whisper.InvalidTimeInterval` every time.  It doesn't seem to be an issue with specific whisper files, because it happens at a different point in the sync every time, and seems to happen regardless of batch size.  Here's the traceback:

    Traceback (most recent call last):
      File "/opt/graphite/bin/carbon-sync", line 9, in <module>
        load_entry_point('carbonate==0.2.2', 'console_scripts', 'carbon-sync')()
      File "/opt/graphite/lib/carbonate-0.2.2-py2.6.egg/carbonate/cli.py", line 197, in carbon_sync
        args.storage_dir, args.rsync_options)
      File "/opt/graphite/lib/carbonate-0.2.2-py2.6.egg/carbonate/sync.py", line 116, in run_batch
        merge_elapsed = sync_batch(metrics_to_heal)
      File "/opt/graphite/lib/carbonate-0.2.2-py2.6.egg/carbonate/sync.py", line 57, in sync_batch
        heal_metric(staging, local)
      File "/opt/graphite/lib/carbonate-0.2.2-py2.6.egg/carbonate/sync.py", line 72, in heal_metric
        fill_archives(source, dest, time())
      File "/opt/graphite/lib/carbonate-0.2.2-py2.6.egg/carbonate/fill.py", line 110, in fill_archives
        fill(src, dst, gapstart - step, start)
      File "/opt/graphite/lib/carbonate-0.2.2-py2.6.egg/carbonate/fill.py", line 72, in fill
        (timeInfo, values) = fetch(src, fromTime, untilTime)
      File "/opt/graphite/lib/whisper.py", line 667, in fetch
        return file_fetch(fh, fromTime, untilTime)
      File "/opt/graphite/lib/whisper.py", line 683, in file_fetch
        raise InvalidTimeInterval("Invalid time interval")
    whisper.InvalidTimeInterval: Invalid time interval

**Comment by jssjr** [comment]
Hi @garrettux! I feel like a :shit: for not getting back to you quickly. Did you ever figure this out?

Out of curiosity, are the clocks on both machines in sync? `whisper-fill` (and by extension, `carbon-sync`) relies on python calling `time.time()` to determine the upper bound for datapoints to sync.

**Comment by mrmanc** [comment]
I added some debug to the local whisper.py file from the [latest version](https://github.com/graphite-project/whisper/blob/master/whisper.py#L766) and can see that there is a comparison between two equal timestamps which is causing the exception. Looking at the logic it looks like there was a flaw in the logic that was fixed in [this commit](https://github.com/graphite-project/whisper/commit/cd389a5f7ba9722e5035703c39027db4fc947ddd#diff-28c419e9b50274d12d6fbaaa79f6d0ce) where it threw an exception for equal timestamps when it looks like what was intended was an error when the `fromTime` is after `untilTime`.

I’ve patched the local version and run again and I have a different error. Progress!
